test/TestExpectations.json

```diff
@@ -1121,6 +1121,34 @@
     "expectations": ["FAIL", "PASS"],
     "comment": "TODO: add a comment explaining why this expectation is required (include links to issues)"
   },
+  {
+    "testIdPattern": "[cookies.spec] Cookie specs BrowserContext.cookies should find cookie created in page",
+    "platforms": ["darwin", "linux", "win32"],
+    "parameters": ["cdp", "firefox"],
+    "expectations": ["FAIL"],
+    "comment": "Firefox CDP does not support Storage.getCookies (UnknownMethodError)"
+  },
+  {
+    "testIdPattern": "[cookies.spec] Cookie specs BrowserContext.cookies should find no cookies in new context",
+    "platforms": ["darwin", "linux", "win32"],
+    "parameters": ["cdp", "firefox"],
+    "expectations": ["FAIL"],
+    "comment": "Firefox CDP does not support Storage.getCookies (UnknownMethodError)"
+  },
+  {
+    "testIdPattern": "[cookies.spec] Cookie specs BrowserContext.setCookie should work",
+    "platforms": ["darwin", "linux", "win32"],
+    "parameters": ["firefox", "webDriverBiDi"],
+    "expectations": ["FAIL"],
+    "comment": "expiry value `-1` not supported by BiDi"
```

Comment:
IIRC -1 denotes undefined expiry. Instead of exposing -1 on the public API, let's accept undefined | uint as expiry and transform to -1 for CDP if needed.

Reply:
e.g., https://github.com/puppeteer/puppeteer/pull/12920/files#diff-5e221f32e7d0b3ccd3f64869c5fc0b1c23e3386d2157dad6f54002ea410b9f8dR246

Reply:
I remembered it the other way around, let's keep -1 on the interface but when calling a WebDriver BiDi endpoint convert -1 to undefined.

Reply:
For reference, the WebDriver BiDi interface https://w3c.github.io/webdriver-bidi/#command-storage-setCookie

Reply:
Okay, I changed it as follows (in bidi/Page.ts):
```ts
...(cookie.expires !== undefined && cookie.expires !== -1 ? {expiry: cookie.expires} : {}),
```
I can technically use something like `!([undefined, -1].includes(cookie.expires))` instead as the condition if you think it's better